libs/eavmlib/src/network.erl

```diff
@@ -314,13 +295,17 @@ stop() ->
 %%-----------------------------------------------------------------------------
 -spec sta_rssi() -> {ok, Rssi :: db()} | {error, Reason :: term()}.
 sta_rssi() ->
-    Port = get_port(),
-    Ref = make_ref(),
-    Port ! {self(), Ref, rssi},
-    receive
-        {Ref, {error, Reason}} -> {error, Reason};
-        {Ref, {rssi, Rssi}} -> {ok, Rssi};
-        Other -> {error, Other}
+    case whereis(network_port) of
+        undefined ->
+            {error, network_down};
+        Pid ->
+            Ref = make_ref(),
+            Pid ! {self(), Ref, rssi},
+            receive
+                {Ref, {error, Reason}} -> {error, Reason};
+                {Ref, {rssi, Rssi}} -> {ok, Rssi};
```

Comment:
What happens when we add ethernet support? This is something we should be doing soon, to support networking for ESP32-P4 and other boards that lack on baord wifi, but may come with ethernet port on-board. This will also conflict with my open PR to allow starting the driver without immediately starting a connection.